File: 0027-remove-element/0027-remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let k = 0;
    let start = 0;
    let end = nums.length-1;

    while(start <= end){
        if(nums[start] === val){
            const extra = nums[start];
            nums[start] = nums[end];
            nums[end] = extra;
            end--;
        }else{
            start++;
            k++;
        }
    }
    return k;
};

File: 0027-remove-element/0027-remove-element.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int k = 0;
        int end = nums.Length - 1;
        int start = 0;

        while(start <= end){
            if(nums[start] == val){
                nums[start] = nums[end];
                nums[end] = val;
                end--;
            }else{
                start++;
                k++;
            }
        }
        return k;
    }
}